game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java

```diff
@@ -76,22 +93,40 @@ private synchronized void loadData() {
       gameData.acquireReadLock();
       try {
         final List<PlayerID> players = getPlayers();
-        final Collection<String> alliances = getAlliances();
-        collectedData = new String[players.size() + alliances.size()][statsResource.length + 1];
+        final Map<String, Set<PlayerID>> allianceMap = getAllianceMap();
+        collectedData = new String[players.size() + allianceMap.size()][resourceStats.size() + 1];
         int row = 0;
+        final Map<PlayerID, IntegerMap<Resource>> resourceIncomeMap = new HashMap<>();
         for (final PlayerID player : players) {
           collectedData[row][0] = player.getName();
-          for (int i = 0; i < statsResource.length; i++) {
-            collectedData[row][i + 1] =
-                statsResource[i].getFormatter().format(statsResource[i].getValue(player, gameData));
+          final IntegerMap<Resource> resourceIncomes = AbstractEndTurnDelegate.findEstimatedIncome(player, gameData);
+          resourceIncomeMap.put(player, resourceIncomes);
+          for (int i = 0; i < resourceStats.size(); i++) {
+            final ResourceStat resourceStat = resourceStats.get(i);
+            final Resource resource = resourceStat.resource;
+            final double quantity = resourceStat.getValue(player, gameData);
+            final StringBuilder text = new StringBuilder(resourceStat.getFormatter().format(quantity) + " (");
```

Comment:
`text` is a bit vague, when we get to the line: ` collectedData[row][i + 1] = text.toString();`; it's a bit hard to understand as the terminology is overly generic. Something like `playerResourceIncome` could work, I am having trouble figuring out exactly which text is going into this variable.

For this code block extracting to a well named private static helper method could help. The append with a potential `()` or `(+)` is hard to understand; I'll need to study this further, but I'm curious if that really is the result output and where that is used.

Reply:
Extracted helper method.